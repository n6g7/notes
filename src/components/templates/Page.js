import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Header } from '@organisms'

const Main = styled.main`
  align-items: center;
  display: flex;
  flex-flow: column nowrap;
`

const Article = styled.article`
  width: 50rem;

  @media (max-width: 60rem) {
    padding: 0 1rem;
    width: 100%;
  }
`

class Page extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired
  }

  render () {
    return <Main {...this.props}>
      <Header />
      <Article>
        {this.props.children}
      </Article>
    </Main>
  }
}

export default Page
