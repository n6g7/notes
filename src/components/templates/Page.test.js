import React from 'react'
import { shallow } from 'enzyme'

import Page from './Page'

describe('<Page />', () => {
  it('renders well', () => {
    const wrapper = shallow(<Page>
      <p>Hi!</p>
    </Page>)
    expect(wrapper).toMatchSnapshot()
  })
})
