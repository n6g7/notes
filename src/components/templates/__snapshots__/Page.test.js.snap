// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> renders well 1`] = `
<styled.main>
  <Header />
  <styled.article>
    <p>
      Hi!
    </p>
  </styled.article>
</styled.main>
`;
